packages/auth/src/authClients/supabase.ts

```diff
@@ -115,9 +115,17 @@ export const supabase = (client: Supabase): AuthClientSupabase => {
       return await client.auth.user()
     },
     restoreAuthState: async () => {
-      await client.auth.getSessionFromUrl()
+      const { data: session } = await client.auth.getSessionFromUrl()
```

Comment:
I think this would still actually be a bug even with the fix because we call restoreAuth anytime you come to the page. So in the use case I identified the user didn't have a session. But if they did have a session I still wouldn't want the params to be removed.

Reply:
I'll let Kris answer this. Just wanted to add a few things meanwhile:

> So, how do you restore when there is already a session?

We can check for the existing session before setting it from the URL. `supabase.auth.session()`
Though, in some edge-case where a user is already logged in with a valid session and tries to re-login (not sure if possible), then we'll ignore newer session detail from URL because we already have session.
> As for how the params get on, is this part of the "redirectTo" in Supabas elogin?

I think so, yes.

@dthyresson @KrisCoulson I'll make changes, let me know if this all sounds okay.